packages/nimble-components/src/select/index.ts

```diff
@@ -622,9 +622,36 @@ export class Select
         // implementation handles skipping non-selected disabled options for the initial
         // selected value.
         this.setSelectedOptions();
+        if (this.open) {
+            this.setActiveOption(this.selectedIndex);
+        }
         this.updateValue();
     }
 
+    /**
+     * @internal
+     * Fork of Listbox implementation, so that the selectedIndex is not changed while the dropdown
+     * is open.
+     */
+    public override typeaheadBufferChanged(_: string, __: string): void {
+        if (this.$fastController.isConnected) {
+            const typeaheadMatches = this.getTypeaheadMatches();
+
+            if (typeaheadMatches.length) {
+                const activeOption = this.options.indexOf(
+                    typeaheadMatches[0] as ListOption
+                );
+                if (activeOption > -1 && !this.open) {
+                    this.selectedIndex = activeOption;
+                } else if (this.filterMode === FilterMode.none) {
+                    this.setActiveOption(activeOption);
+                }
```

Comment:
Should this `else if` block check `this.open` and only call `this.setActiveOption` if the select is open? I'm guessing there isn't any harm in calling `this.setActiveOption` when it is closed, but it would probably make things more clear & consistent if we aren't updating `this.setActiveOption` if half of the code paths when the select is closed but not the other half.

Reply:
As discussed in person, we _do_ call `setActiveOption` for code paths when the dropdown is closed. This really meant that this implementation could simplify to unconditionally calling `setActiveOption`.